Userland/Services/ImageDecoder/ConnectionFromClient.h

```diff
@@ -24,9 +25,17 @@ class ConnectionFromClient final
     virtual void die() override;
 
 private:
+    using Job = Function<void()>;
+
     explicit ConnectionFromClient(NonnullOwnPtr<Core::LocalSocket>);
 
     virtual Messages::ImageDecoderServer::DecodeImageResponse decode_image(Core::AnonymousBuffer const&, Optional<Gfx::IntSize> const& ideal_size, Optional<ByteString> const& mime_type) override;
+    virtual void cancel_decoding(i64 image_id) override;
+
+    Job make_decode_image_job(i64 image_id, Core::AnonymousBuffer, Optional<Gfx::IntSize> ideal_size, Optional<ByteString> mime_type);
+
+    IDAllocator m_id_allocator;
+    HashMap<i64, Job> m_pending_jobs;
```

Comment:
For `i64`, I think a monotonically increasing `i64` is strictly better than using `IDAllocator`. Trivial implementation, and no ID will ever be reused within one program execution.